app/routes.py

```diff
@@ -1,2 +1,200 @@
+                })
+            return jsonify(tasks_list)
+            
+        elif request.method == 'POST':
+            request_body = request.get_json()
+            if 'title' not in request_body.keys() or 'description' not in request_body.keys() \
+                                                  or 'completed_at' not in request_body.keys():
+                return make_response({"details": "Invalid data"}, 400)
+            else:
+                new_task = Task(title=request_body['title'],
+                                description=request_body['description'],
+                                completed_at= request_body['completed_at'] if not request_body['completed_at'] \
+                                                else datetime.utcnow()
+                                )
+                db.session.add(new_task)
+                db.session.commit()
+                return make_response({"task": { "id": new_task.task_id,
+                                                "title": new_task.title,
+                                                "description": new_task.description,
+                                                "is_complete": False if new_task.completed_at is None else True
+                                              }}, 201)
+
+@tasks_bp.route('/<task_id>', methods=['GET', 'PUT', 'DELETE'])
+def handle_one_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return jsonify(task), 404
+    one_task = {"task": {"id": task.task_id,
+                            "title": task.title,
+                            "description": task.description,
+                            "is_complete": False if task.completed_at is None else task.completed_at
+                        }}
+    if request.method == 'GET' and task.goal_id is None:
+        return one_task
+    elif request.method == 'GET':
+        one_task['task']['goal_id'] = task.goal_id
+        return one_task
+    elif request.method == 'PUT':
+        updates = request.get_json()
+        task.title = updates['title']
+        task.description = updates['description']
+        if 'completed_at' in updates.keys():
+            task.completed_at = updates['completed_at'] if not updates['completed_at'] \
+                                else datetime.utcnow()
+        
+        db.session.commit()
+        return make_response({"task": { "id": task.task_id,
+                                    "title": task.title,
+                                    "description": task.description,
+                                    "is_complete": False if task.completed_at is None else True
+                                    }})
+    elif request.method == 'DELETE':
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f"Task {task.task_id} \"{task.title}\" successfully deleted"})
+
+@tasks_bp.route('/<task_id>/mark_complete', methods=['PATCH'])
+def mark_complete(task_id):
```

Comment:
You can combine complete/ incomplete here with a conditional. Here is an example: 
`
 if task.completed_at:
        data = {
            "channel": "general",
            "text": f"Task {task.title} has been marked complete",
        }
    else: 
        data = {
            "channel": "general",
            "text": f"Task {task.title} has been marked incomplete",
        }
`